app/javascript/menu/group-switcher.jsx

```diff
@@ -0,0 +1,55 @@
+import Dropdown from 'carbon-components-react/es/components/Dropdown';
+import React from 'react';
+import SideNavItem from 'carbon-components-react/es/components/UIShell/SideNavItem';
+import { Collaborate20 } from '@carbon/icons-react';
+
+const { miqChangeGroup } = window;
+
+export const GroupSwitcher = ({ miqGroups, currentGroup, expanded }) => {
+  const options = miqGroups.map((g) => ({
+    label: g.description,
+    value: g.id,
+  }));
+
+  const currentOption = {
+    label: currentGroup.description,
+    value: currentGroup.id,
+  };
+
+  const groupChange = ({ selectedItem }) => {
+    const group_id = selectedItem.value;
+    if (group_id && group_id !== currentGroup.id) {
+      miqChangeGroup(group_id);
+    }
+  };
+
+  return (
+    <div
+      className="menu-group"
+      title={`${__("Current group:")} ${currentOption.label}`}
+    >
+      { expanded ? (
```

Comment:
This kind of combination of ternary operators is super hard to follow even when we're not doing it in react, can you use something cleaner please?

Reply:
split into singleGroup / multiGroup / collapsed variables